File: java/problem-006/original/Main.java
public class Main {
    public static void main(String[] args) {
        int[] firstHundred = numbersBetween(1, 100);
        int answer = squareOfSum(firstHundred) - sumOfSquares(firstHundred);

        System.out.println(answer);
    }

    public static int[] numbersBetween(int start, int end) {
        int[] numbers = new int[end - start + 1];

        for(int i = start; i <= end; i++) {
            int index = i - start;
            numbers[index] = i;
        }

        return numbers;
    }

    private static int squareOfSum(int[] nums) {
        int sum = 0;

        for (int num : nums) {
            sum += num;
        }

        return (sum * sum);
    }

    private static int sumOfSquares(int[] nums) {
        int sum = 0;

        for (int num : nums) {
            int square = (num * num);
            sum += square;
        }

        return sum;
    }
}

File: go/problem-006/original/main.go
package main

import (
    "fmt"
)

func main() {
    firstHundred := numbersBetween(1, 100)
    answer := squareOfSum(firstHundred) - sumOfSquares(firstHundred)

    fmt.Println(answer)
}

func numbersBetween(start int, end int) []int {
    numNumbers := end - start + 1
    numbers := make([]int, numNumbers, numNumbers)

    for i, n := 0, start; n <= end; i, n = i+1, n+1 {
        numbers[i] = n
    }

    return numbers
}

func squareOfSum(nums []int) int {
    var sum int

    for _, num := range nums {
        sum += num
    }

    return (sum * sum)
}

func sumOfSquares(nums []int) int {
    var sum int

    for _, num := range nums {
        square := num * num
        sum += square
    }

    return sum
}
